### ISSUE TITLE
Feature Request / Papercut: Improve Lazy Plugin Loading

### ISSUE THREAD

**Issue by JMurph2015** [syl20bnr/spacemacs#10574]
A nice feature SpaceVim has down pat that Spacemacs has trouble with is lazy loading of packages and layers.  For my current config, I'm pulling in >300 packages on startup, which is a bit time intensive (few seconds to startup), though not beyond papercut status.  

Potentially, I have more layers explicitly listed in my .spacemacs than is necessary, in which case does anyone have advice on pitfalls there?  The system works extremely well elsewhere, so it'd be nice to have Spacemacs taking better advantage of it.

**Comment by CeleritasCelery** [comment]
Spacemacs **does** lazy load most packages. Also the maintainers have been working on improving startup times for the next release. If you find yourself opening and closing Spacemacs a lot consider using the daemon. See [this](https://medium.com/@bobbypriambodo/blazingly-fast-spacemacs-with-persistent-server-92260f2118b7) article

**Comment by NightMachinary** [comment]
There was a commit that lazy-loaded by default. They reverted it because it was breaking the API. Use a daemon as CeleritasCelery has suggested. It works great.

**Comment by JMurph2015** [comment]
@CeleritasCelery I know that lazy-loading is supported, but considering I'm loading 300+ packages on startup (even before I've opened a file), I'd say this still has a long way to go before we can call it a victory.

@NightMachinary 

* I am aware and do use the daemon pretty often, but there are loads of use cases where the daemon comes up short.  
    * For instance, it's really clumsy if you happen to need more than one running instance of emacs at a time (potentially one on a virtual console and one on a X11 desktop, or just two on different workspaces in an X11 desktop).
  * The other big deal breaker is that the daemon is essentially unusable in situations where you need elevated privileges (ie modifying a config file).  

So for those reasons, I pretty frequently end up having to start and restart Emacs.  Also, it's just absurd for a text editor to take 5+ seconds to load on a quad-core Skylake i7 with an SSD, I mean that's getting into Matlab territory.

**Comment by CeleritasCelery** [comment]
> I know that lazy-loading is supported, but considering I'm loading 300+ packages on startup (even before I've opened a file), I'd say this still has a long way to go before we can call it a victory.

You are not "loading" 300 packages. The majority of them are only loading the autoloads and will be loaded later. You can use `featurep` to see which packages are actually loaded.

> For instance, it's really clumsy if you happen to need more than one running instance of emacs at a time (potentially one on a virtual console and one on a X11 desktop, or just two on different workspaces in an X11 desktop).

You can have multiple frame with the daemon so you can have one in each console or desktop. Also even if you did have multiple Emacs they can all be daemon, so you don't need to really ever close them. 

> The other big deal breaker is that the daemon is essentially unusable in situations where you need elevated privileges (ie modifying a config file).

that is totally possible with daemon. See [this](https://stackoverflow.com/questions/10190358/is-there-some-way-to-open-file-with-emacsclient-user-but-with-root-permission)
